http_server.py

```diff
@@ -0,0 +1,107 @@
+import socket
+import os
+import sys
+
+
+ROOT = "./webroot/"
+ADDR = ("127.0.0.1", 8000)
+server = socket.socket(
+    socket.AF_INET, socket.SOCK_STREAM, socket.IPPROTO_IP
+)
+server.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+
+
+def resolve_uri(uri):
+    if "../" in uri:
+        raise UserWarning("403 Forbidden")
+
+    if os.path.isfile(uri):
+        TYPE = uri.split(".")[-1].lower()
+        if TYPE in ('PNG', 'JPG', 'JPEG', 'GIF'):
+            TYPE = 'image/' + TYPE
+        else:
+            TYPE = 'text/' + TYPE
```

Comment:
You're pretty limited in what type of file you can return.  For example, the python source file will have a hard time working this way.

Look at the `mimetypes` module from the python standard library.  That will do this job much better and much more thoroughly.